spec/data-types-and-values.html

```diff
@@ -7,12 +7,63 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>
+        The Record type is the set of all finite mappings from Strings to ECMAScript primitive values including Record and Tuple. Each record value holds an associated [[Fields]] List value which is a list of pairs of the form { [[Key]], [[Value]] } where the [[Key]] is a String and [[Value]] is any primitive value. Entries of [[Fields]] are sorted by [[Key]] in code unit order. The [[Fields]] List and its entries are never modified.
+     </p>
+
+      <emu-clause id="sec-makerecordvalue" aoid="MakeRecordValue">
+        <h1>MakeRecordValue ( _entries_ )</h1>
+        <p>This abstract operation MakeRecordValue takes a _entries_ argument. It is used to specify the creation of new Record value types. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: _entries_ is a List.
+            1. Let _uniqueEntries_ be an empty Set.
+            1. For each _kv_ in _entries_:
+              1. Assert: _kv_.[[Key]] is a String.
+              1. Assert: _kv_.[[Value]] is a primitive type.
+              1. For each _existing_ in _uniqueEntries_:
+                1. If SameValue(_existing_.[[Key]], _kv_.[[Key]]
+                  1. Remove _existing_ from _uniqueEntries_
+                  1. Break
+              1. Add _kv_ to _uniqueEntries_
+            1. Let _sortedEntries_ be a new List containing the same values as the list _entries_ where the values are ordered as if an Array of the same values had been sorted using `Array.prototype.sort` using <emu-xref href="#sec-recordentrycompare">RecordEntryCompare</emu-xref> as _comparefn_.

```

Comment:
Pedantically speaking, it seems inappropriate to use Array.prototype.sort on a List of Records--you can't put these Records in an Array. Also I think it would be clearer to use less indirection in this algorithm (though I can see how this was inspired by wording in module namespace exotic objects). How about this wording:

1. Sort _entries_ such that, for any entry two entries _a_ preceding _b_, then the result of performing Abstract Relational Comparison _a_.[[Key]] < _b_.[[Key]] is *true*.

Reply:
Got my inspiration from here: https://tc39.es/ecma262/#sec-modulenamespacecreate

I understand it looks heavy but this seems to do what we want. I'll explore Abstract Relational Comparison later once I'm done with the rest.